.github/workflows/GNU.yml

```diff
@@ -83,7 +104,7 @@ jobs:
         GNULIB_DIR="${PWD}/gnulib"
         pushd gnu
 
-        unbuffer timeout -sKILL 3600 make -j "$(nproc)" check SUBDIRS=. RUN_EXPENSIVE_TESTS=yes RUN_VERY_EXPENSIVE_TESTS=yes VERBOSE=no || :
+        timeout -sKILL 4h make -j "$(nproc)" check SUBDIRS=. RUN_EXPENSIVE_TESTS=yes RUN_VERY_EXPENSIVE_TESTS=yes VERBOSE=no || : # Kill after 4 hours in case something gets stuck in make
```

Comment:
4h ? :)
1h seems enough, no ?

Reply:
not really, run time for the 'Run GNU tests' stage was 1 hour 9 minutes on the last commit before I opened the PR (adding some shorter timeouts on a few more tests known to hang could speed things up a bit). I'm not worried about making this as sort as possible, with the timeout being applied per-test this was just left as a last resort in case something in make gets jammed. 
It turns out there's a built in 6 hour limit on github actions, so we could also take this out and rely on that if something does get break.

Reply:
Maybe 2 hours then. Also, `expect` can now be removed from the `apt-get install` line above, since you are removing the `unbuffer` command.